src/services/__tests__/pagoPAProxyService.test.ts

```diff
@@ -177,6 +182,24 @@ describe("PagoPAProxyService#getPaymentInfo", () => {
     expect(res.kind).toEqual("IResponseErrorValidation");
   });
 
+  it("getPaymentInfoDefaultDecoder must validate correctly a PaymentProblemJson response", async () => {
+    const a = {
+      detail: PaymentFaultEnum.DOMAIN_UNKNOWN
+    };
+    const validator = getPaymentInfoDefaultDecoder();
+    const res = mockRes();
+    // tslint:disable: no-object-mutation
+    res.body = a;
+    res.status = 500;
+    // tslint:enable: no-object-mutation
+    const validation = await validator(res);
```

Comment:
does `validator` return a promise?

Reply:
Yes it does. `export declare type ResponseDecoder<R> = (response: Response) => Promise<t.Validation<R> | undefined>;`